pyuvdata/uvflag/tests/test_uvflag.py

```diff
@@ -1655,39 +1700,41 @@ def test_set_lsts(uvf_from_data, background):
     assert uvf2._lst_array == uvf._lst_array
 
 
-@pytest.mark.filterwarnings("ignore:Nants_telescope, antenna_")
 def test_set_telescope_params(uvdata_obj):
     uvd = uvdata_obj
-    uvd.set_telescope_params(overwrite=True)
+    uvd.set_telescope_params(overwrite=True, warn=False)
     ants_with_data = np.union1d(uvd.ant_1_array, uvd.ant_2_array)
     uvd2 = uvd.select(
         antenna_nums=ants_with_data[: uvd.Nants_data // 2],
         keep_all_metadata=False,
         inplace=False,
     )
     uvf = UVFlag(uvd2, use_future_array_shapes=True)
-    assert uvf._antenna_names == uvd2._antenna_names
-    assert uvf._antenna_numbers == uvd2._antenna_numbers
-    assert uvf._antenna_positions == uvd2._antenna_positions
+    assert uvf.telescope._antenna_names == uvd2.telescope._antenna_names
```

Comment:
So it turns out that the telescope objects aren't equal because they have different sets of required parameters (UVData's requires `instrument` while UVFlag's does not). So I can make copies of the telescope objects, adjust those requirements and then test that the telescopes are equal or I can just test these attributes.